checkout/payment_flow.py

```
@@ -0,0 +1,63 @@
+    def checkout(self, order_id, method):
+        order = self.load_order(order_id)
+        if order["status"] == "PAID":
+            return {"ok": False, "error": "already paid"}
+
+        key = self.make_idempotency_key(order_id, method)
+        logger.info("initiating payment key=%s", key)
+
+        response = self.payments.initiate(order_id, method)
+        self.db.execute(
+            "UPDATE orders SET payment_ref = '" + response["ref"] + "' WHERE id = "
+            + str(order_id)
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)